mongocursorpagination/mongo_find.go

```diff
@@ -0,0 +1,349 @@
+// Package official mongo driver, eases the computation of pagination information
+// of a find mongo query by augmenting the base query with cursor information and
+// returning a cursor.
+package mongocursorpagination
```

Comment:
Could you split `mgo` and `mongo` into two different packages? This is so a consumer does not have to vendor both drivers. So `mongocursorpagination/mgo` and `mongocursorpagination/mongo`